flask/sessions.py

```diff
@@ -423,3 +426,16 @@ def save_session(self, app, session, response):
             path=path,
             secure=secure
         )
+
+    def _patch_vary_cookie_header(self, response):
+        """
+        Add a "Vary: Cookie" header, preserving the original header if one is present.
+
+        The header should retain it's original ordering.
+        """
+        vary_headers = self.comma_delim_header_re.split(response.headers.get('Vary', ''))
+        if 'Cookie' not in vary_headers:
```

Comment:
Need to be case insensitive:

~~~python
if not any(h.lower() == 'cookie' for h in headers)
~~~